src/java/org/apache/cassandra/config/Config.java

```diff
@@ -910,7 +910,11 @@ public static void setClientMode(boolean clientMode)
     public volatile Map<StartupCheckType, Map<String, Object>> startup_checks = new HashMap<>();
 
     public volatile DurationSpec.LongNanosecondsBound repair_state_expires = new DurationSpec.LongNanosecondsBound("3d");
-    public volatile int repair_state_size = 100_000;
+
+    // Only one of repair_state_size and repair_state_heap_size should be set
+    @Deprecated
+    public volatile Integer repair_state_size = null;
+    public volatile DataStorageSpec.IntBytesBound repair_state_heap_size = new DataStorageSpec.IntBytesBound(5, DataStorageSpec.DataStorageUnit.MEBIBYTES);
```

Comment:
spoke in slack, this is a bug as we can not tell if the user set `5mb` or not... 

here is the semantic we would want in yamll

```
repair_state_size: 42
```

we are size based

```
repair_state_heap_size: 10mb
```

we are heap based

```
repair_state_size: 42
repair_state_heap_size: 10mb
```

reject

Reply:
Addressed - thanks.